src/Compilers/CSharp/Test/Emit2/Semantics/LockTests.cs

```diff
@@ -0,0 +1,1797 @@
+            // lock (l) { }
+            Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "l").WithArguments("System.Threading.Lock+Scope", "Dispose").WithLocation(2, 7));
+    }
+
+    [Fact]
+    public void ScopeDisposeTakesParameters()
+    {
+        var source = """
+            System.Threading.Lock l = new();
+            lock (l) { }
+
+            namespace System.Threading
+            {
+                public class Lock
+                {
+                    public Scope EnterLockScope() => new Scope();
+
+                    public ref struct Scope
+                    {
+                        public void Dispose(int x) { }
+                    }
+                }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (2,7): error CS0656: Missing compiler required member 'System.Threading.Lock+Scope.Dispose'
+            // lock (l) { }
+            Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "l").WithArguments("System.Threading.Lock+Scope", "Dispose").WithLocation(2, 7));
+    }
+
+    [Fact]
+    public void ScopeDisposeTakesParameters_Optional()
+    {
+        var source = """
+            System.Threading.Lock l = new();
+            lock (l) { }
+
+            namespace System.Threading
+            {
+                public class Lock
+                {
+                    public Scope EnterLockScope() => new Scope();
+
+                    public ref struct Scope
+                    {
+                        public void Dispose(int x = 1) { }
+                    }
+                }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (2,7): error CS0656: Missing compiler required member 'System.Threading.Lock+Scope.Dispose'
+            // lock (l) { }
+            Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "l").WithArguments("System.Threading.Lock+Scope", "Dispose").WithLocation(2, 7));
+    }
+
+    [Fact]
+    public void ScopeDisposeTakesParameters_ParamsArray()
+    {
+        var source = """
+            System.Threading.Lock l = new();
+            lock (l) { }
+
+            namespace System.Threading
+            {
+                public class Lock
+                {
+                    public Scope EnterLockScope() => new Scope();
+
+                    public ref struct Scope
+                    {
+                        public void Dispose(params int[] xs) { }
+                    }
+                }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (2,7): error CS0656: Missing compiler required member 'System.Threading.Lock+Scope.Dispose'
+            // lock (l) { }
+            Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "l").WithArguments("System.Threading.Lock+Scope", "Dispose").WithLocation(2, 7));
+    }
+
+    [Fact]
+    public void InternalLock()
```

Comment:
>InternalLock [](http://example.com/codeflow?start=16&length=12)

I am not sure what was the intent for this test. It looks like the `Lock` type isn't declared internal